### DISCUSSION TITLE
Using WikibaseIntegrator for a Wikibase Docker instance

### DISCUSSION THREAD

**Discussion by Hildepont-0** [LeMyst/WikibaseIntegrator#88] [Q&A] [answered]
Hi everyone,
I try to use WikibaseIntegrator such as in the first example (with P24 a string property) : 

> from wikibaseintegrator import wbi_core, wbi_login
> login_instance = wbi_login.Login(user='WikibaseAdmin@bot_1', pwd='my pswd',  mediawiki_api_url='http://localhost:8181/api.php', mediawiki_index_url='http://localhost:8181')
> entrez_gene_id = wbi_core.String(value='text', prop_nr='P24')
> data = [entrez_gene_id]
> wd_item = wbi_core.ItemEngine(data=data)
> wd_item.write(login_instance)

And I get an error, typing the last code line : 

> Error while writing to the Wikibase instance
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/popa/Code/tools/WikibaseIntegrator/WikibaseIntegrator/wikibaseintegrator/wbi_core.py", line 673, in write
>     raise MWApiError(json_data)
> wikibaseintegrator.wbi_core.MWApiError: {'error': {'code': 'badtoken', 'info': 'Invalid CSRF token.', '*': 'See https://www.wikidata.org/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes.'}, 'servedby': 'mw1285'}

I think I made a mistake in the login parameters but I'm not sure. Is this obvious for someone ?

**Comment by Hildepont-0** [comment]
The error changed...

> Error while writing to the Wikibase instance
> Traceback (most recent call last):
>   File "integrator.py", line 10, in <module>
>     wd_item.write(login_instance)
>   File "/home/popa/.local/lib/python3.8/site-packages/wikibaseintegrator/wbi_core.py", line 1010, in write raise MWApiError(json_data)
> wikibaseintegrator.wbi_core.MWApiError: {'error': {'code': 'badtoken', 'info': 'Invalid CSRF token.', '*': 'See https://www.wikidata.org/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes.'}, 'servedby': 'mw1346'}

**Comment by LeMyst** [comment]
Hello @Hildepont-0 
Do you use the version from the git (with a git clone) or did you install it from 'pip'?

**Reply by Hildepont-0** [reply]
from the git... should I try to reinstall ?

**Reply by LeMyst** [reply]
I can reproduce the issue on my test environnement. I'll fix it. Thanks for reporting.

**Reply by Hildepont-0** [reply]
thanks to you !

**Comment by LeMyst** [comment, marked as answer]
Ok, I understood your issue.
You are logged to your instance but you try to edit/access Wikidata (the default instance).

Try to add this at the top of your script and remove the mediawiki_api_url and mediawiki_index_url from your wbi_login.Login():
```
from wikibaseintegrator.wbi_config import config as wbi_config

wbi_config['MEDIAWIKI_API_URL'] = 'http://localhost:8181/api.php'
wbi_config['SPARQL_ENDPOINT_URL'] = 'http://localhost:8989/bigdata/sparql'
wbi_config['WIKIBASE_URL'] = 'http://wikibase.svc'
```

**Comment by Hildepont-0** [comment]
I  get a warning but it works ! :)
> /usr/local/lib/python3.8/dist-packages/wikibaseintegrator-0.10.0.dev0-py3.8.egg/wikibaseintegrator/wbi_core.py:1412: UserWarning: Warning: No distinct value properties found
> Please set P2302 and Q21502410 in your Wikibase or set `core_props` manually.
> Continuing with no core_prop

Merci @LeMyst
